src/unix/linux/disk.rs

```diff
@@ -177,38 +197,69 @@ fn get_actual_device_name(device: &OsStr) -> String {
         .unwrap_or_default()
 }
 
-fn new_disk(
-    device_name: &OsStr,
+unsafe fn load_statvfs_values(
     mount_point: &Path,
-    file_system: &OsStr,
-    removable_entries: &[PathBuf],
-    procfs_disk_stats: &HashMap<String, DiskStat>,
-) -> Option<Disk> {
-    let mount_point_cpath = to_cpath(mount_point);
-    let type_ = find_type_for_device_name(device_name);
-    let mut total = 0;
-    let mut available = 0;
-    let mut is_read_only = false;
-    unsafe {
+    refresh_kind: DiskRefreshKind,
+) -> Option<(u64, u64, bool)> {
+    if refresh_kind.details() {
+        let mount_point_cpath = to_cpath(mount_point);
         let mut stat: statvfs = mem::zeroed();
         if retry_eintr!(statvfs(mount_point_cpath.as_ptr() as *const _, &mut stat)) == 0 {
             let bsize = cast!(stat.f_bsize);
             let blocks = cast!(stat.f_blocks);
             let bavail = cast!(stat.f_bavail);
-            total = bsize.saturating_mul(blocks);
-            available = bsize.saturating_mul(bavail);
-            is_read_only = (stat.f_flag & libc::ST_RDONLY) != 0;
+            let total = bsize.saturating_mul(blocks);
+            let available = bsize.saturating_mul(bavail);
+            let is_read_only = (stat.f_flag & libc::ST_RDONLY) != 0;
+
+            if total == 0 {
+                None
+            } else {
+                Some((total, available, is_read_only))
+            }
+        } else {
+            None
         }
-        if total == 0 {
-            return None;
+    } else {
+        Some((0, 0, false))
+    }
```

Comment:
Let's remove some indent. :)

```suggestion
    if !refresh_kind.details() {
        return None;
    }
    let mount_point_cpath = to_cpath(mount_point);
    let mut stat: statvfs = mem::zeroed();
    if retry_eintr!(statvfs(mount_point_cpath.as_ptr() as *const _, &mut stat)) == 0 {
        let bsize = cast!(stat.f_bsize);
        let blocks = cast!(stat.f_blocks);
        let bavail = cast!(stat.f_bavail);
        let total = bsize.saturating_mul(blocks);
        if total == 0 {
            return None;
        }
        let available = bsize.saturating_mul(bavail);
        let is_read_only = (stat.f_flag & libc::ST_RDONLY) != 0;

        Some((total, available, is_read_only))
    } else {
        None
    }
```

Reply:
Nice, done.